client/asset/btc/spv_wrapper.go

```diff
@@ -0,0 +1,1677 @@
+	if blockHash == nil {
+		blockHash, _ = w.mainchainBlockForStoredTx(txHash)
+	}
+
+	// Our last option is neutrino.
+	w.log.Tracef("swapConfirmations - scanFilters: %v:%d (block %v, start time %v)",
+		txHash, vout, blockHash, startTime)
+	utxo, err := w.scanFilters(txHash, vout, pkScript, startTime, blockHash)
+	if err != nil {
+		return 0, false, err
+	}
+
+	if utxo.spend == nil && utxo.blockHash == nil {
+		if assumedMempool {
+			w.log.Tracef("swapConfirmations - scanFilters did not find %v:%d, assuming in mempool.",
+				txHash, vout)
+			// NOT asset.CoinNotFoundError since this is normal for mempool
+			// transactions with an SPV wallet.
+			return 0, false, nil
+		}
+		return 0, false, fmt.Errorf("output %s:%v not found with search parameters startTime = %s, pkScript = %x",
+			txHash, vout, startTime, pkScript)
+	}
+
+	if utxo.blockHash != nil {
+		bestHeight, err := w.getChainHeight()
+		if err != nil {
+			return 0, false, fmt.Errorf("getBestBlockHeight error: %v", err)
+		}
+		confs = uint32(bestHeight) - utxo.blockHeight + 1
+	}
+
+	if utxo.spend != nil {
+		// In the off-chance that a spend was found but not the output itself,
+		// confs will be incorrect here.
+		// In situations where we're looking for the counter-party's swap, we
+		// revoke if it's found to be spent, without inspecting the confs, so
+		// accuracy of confs is not significant. When it's our output, we'll
+		// know the block and won't end up here. (even if we did, we just end up
+		// sending out some inaccurate Data-severity notifications to the UI
+		// until the match progresses)
+		return confs, true, nil
+	}
+
+	// unspent
+	return confs, false, nil
+}
+
+func (w *spvWallet) locked() bool {
+	return w.wallet.Locked()
+}
+
+func (w *spvWallet) walletLock() error {
+	w.wallet.Lock()
+	return nil
+}
+
+func (w *spvWallet) walletUnlock(pw []byte) error {
+	return w.Unlock(pw)
+}
+
+func (w *spvWallet) getBlockHeader(blockHash *chainhash.Hash) (*blockHeader, error) {
+	hdr, err := w.cl.GetBlockHeader(blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	tip, err := w.cl.BestBlock()
+	if err != nil {
+		return nil, fmt.Errorf("BestBlock error: %v", err)
+	}
+
+	blockHeight, err := w.cl.GetBlockHeight(blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return &blockHeader{
+		Hash:          hdr.BlockHash().String(),
+		Confirmations: int64(confirms(blockHeight, tip.Height)),
+		Height:        int64(blockHeight),
+		Time:          hdr.Timestamp.Unix(),
+	}, nil
+}
+
+func (w *spvWallet) getBestBlockHeader() (*blockHeader, error) {
+	hash, err := w.getBestBlockHash()
+	if err != nil {
+		return nil, err
+	}
+	return w.getBlockHeader(hash)
+}
+
+func (w *spvWallet) logFilePath() string {
+	return filepath.Join(w.dir, logDirName, logFileName)
+}
+
+// connect will start the wallet and begin syncing.
+func (w *spvWallet) connect(ctx context.Context, wg *sync.WaitGroup) (err error) {
+	w.wallet = w.newBTCWallet(w.dir, w.cfg, w.chainParams, w.log)
```

Comment:
I think we should call the constructor in the `spvWallet` constructor (`btc.OpenSPVWallet`) and eliminate this `newBTCWallet` field.


```go
	spvw.wallet = walletConstructor(spvw.dir, spvw.cfg, spvw.chainParams, spvw.log)

	btc.node = spvw

	return &ExchangeWalletSPV{
```